### PR TITLE
Monorepo Updates: Yarn, Lerna Fixed, Node v12, etc

### LINKED ISSUE #1040: Change lerna to fixed version mode
Related to https://github.com/pattern-lab/patternlab-node/issues/1039

for simplicity's sake we should consider abandoning lerna independent mode and affixing our versioning of the entire ecosystem at a time. so core@4 works with cli@4 works with @pattern-lab/plugin-foo@4 etc.

i like independent versioning because i believe in semver but it's clearly causing confusion

### PR DESCRIPTION
**Pull request by sghoweri**
## Summary
- Switches from using NPM to Yarn while doing local PL development
- Switches Lerna versioning from Independent to fixed
- Updated Node to v12
- Removes pre-built UIKit assets from the repo and instead auto-builds when publishing and/or bootstrapping while doing local development
- Speeds up Travis CI / tests by ~1 to 1.5 minutes! 🎉

Closes #1040

### CODE REVIEW

**Comment by bmuenzenmeyer** [comment]
@sghoweri this is incredible! Thank you so much for taking on so much.  Will review in earnest this weekend. 

Onward!

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-10.0.0
+12.9.0
```
Comment (bmuenzenmeyer) [inline .nvmrc]: a small nit for next time - so we're on the same page - i like to adhere to the guidance that production apps should use `ACTIVE`, not `CURRENT` according to this schedule. In this case, we are talking a few days, but it's wanted to bring attention to it.

Comment (sghoweri) [reply .nvmrc]: Totally fair — happy to bump to down to 11.x (but yep, 9 days to go till LTS!). 

LMK if you think we should change this before merging down.

**Comment by bmuenzenmeyer** [comment]
No it's totally fine, a couple days doesn't matter.  And odd numbers are
usually for developing new node features so I tend to stay away too.

**Comment by bmuenzenmeyer** [comment]
very excited for this - but i might not get to it til next week - vacation soon

**Comment by bmuenzenmeyer** [comment]
@sghoweri !!!!

**Comment by bmuenzenmeyer** [comment]
I took a survey of existing monorepo packages, to better understand if we should change our `version` field in `lerna.json`

```
@pattern-lab/cli@1.0.4
@pattern-lab/core@3.0.5
create-pattern-lab@1.0.12
@pattern-lab/edition-node@2.0.7
@pattern-lab/edition-node-gulp@2.0.10
@pattern-lab/edition-twig@3.1.9
@pattern-lab/engine-handlebars@2.0.2
@pattern-lab/engine-liquid@1.0.0-beta.1
@pattern-lab/engine-mustache@2.0.1
@pattern-lab/engine-nunjucks@0.2.1
@pattern-lab/engine-react@0.2.1-beta.1
@pattern-lab/engine-twig@0.2.1-beta.1
@pattern-lab/engine-twig-php@3.0.6
@pattern-lab/engine-underscore@2.0.0-beta.1
@pattern-lab/live-server@1.3.4
```

**Comment by bmuenzenmeyer** [comment]
@sghoweri curious if you have time to read the above and concur before next release

**Comment by sghoweri** [comment]
@bmuenzenmeyer taking a look!!

**Comment by sghoweri** [comment]
@bmuenzenmeyer see answers below:

| package | version | comment | 
| ------- | ------ | ------ | 
| engine-mustache | 2.0.1 | evaluate for deprecation | 
| engine-twig | 0.2.1-beta.1 | evaluate for deprecation |
| engine-underscore | 2.0.0-beta.1 | deprecate soon | 
| engine-nunjucks | 0.2.1 | promote off of 0. | 
| engine-react | 0.2.1-beta.1 | icebox | 
| engine-liquid | 1.0.0-beta.1 | promote out of beta | 

👍

I'd love to see us keep working on the React engine (and perhaps that work opens the door to other JS-driven options) but yeah, it needs more work. Perhaps update the React engine package.json to be marked as private (temporarily) until we're ready to push out further updates?

| package | version | comment | 
| ------- | ------ | ------ | 
| live-server | 1.3.4 | evaluate replacement with direct third party | 

Yes please!! For example, [Browser Sync](https://www.browsersync.io/) could be an excellent off the shelf replacement!

> judging from this - i think we need to move everything up to 4.1.1 ?

I'm inclined to say `5.0.0`, given that we want to get all of our packages synced to a single major version (hard SEMVER reset). Thoughts?

> https://github.com/pattern-lab/starterkit-handlebars-vanilla should really move into the monorepo, with this linked repo and anything else still hanging around independently either archived or outright deleted

Yes please!

**Comment by bmuenzenmeyer** [comment]
> I'm inclined to say `5.0.0`, given that we want to get all of our packages synced to a single major version (hard SEMVER reset). Thoughts?

yeah i am fine with either - i just didnt know if manually setting it to 5 would increment it to 6 upon release. i suppose there is one way to find out. lerna has dry run capability, IIRC

> Yes please!! For example, [Browser Sync](https://www.browsersync.io/) could be an excellent off the shelf replacement!

we used to use that but i was convinced to move away from it by a former contributor. anything that makes our life easier, meets the needs of users, and is secure, I am for



> Perhaps update the React engine package.json to be marked as private (temporarily) until we're ready to push out further updates?

that's a good idea

**Pull request merged**